### PR TITLE
DAOS-7259 dfs: dfs does not require fuse3 to build.

### PR DESCRIPTION
**Pull request by mchaarawi**
also remove dfuse_hl.

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

### CODE REVIEW

File: src/client/dfs/SConscript [hunk 1]
```diff
@@ -46,18 +46,17 @@ def scons():
...
     denv.Install('$PREFIX/lib64/', dfs_internal)
 
+    duns = daos_build.library(denv, 'duns', 'duns.c', LIBS=libraries)
+    denv.Install('$PREFIX/lib64/', duns)
+
     fenv = denv.Clone()
     prereqs.require(fenv, 'fuse')
     dfuse_hl_src = ['dfuse_hl.c']
+    libraries += ['fuse3']
     dfuse_hl = daos_build.program(fenv, 'dfuse_hl', dfuse_hl_src,
```
Comment (jolivier23) [inline src/client/dfs/SConscript:58]: this line is still building dfuse_hl, no?

Also, you need to remove dfuse_hl from utils/rpms/daos.spec and debian (and add change log)

**Review by jolivier23 — CHANGES_REQUESTED** [review]

File: src/client/dfs/SConscript (hunk 1 above)
Comment (mchaarawi) [reply src/client/dfs/SConscript:58]: yea i had removed then re-added it.
i want to build it but not install it.
i see rpm build failing, thanks for the hint.

Comment (mchaarawi) [reply src/client/dfs/SConscript:58]: i guess maybe just not build it is also fine.. sorry im going back and forth on this.

Comment (johannlombardi) [reply src/client/dfs/SConscript]: I would be in favor of removing it ...

**Review by jolivier23 — APPROVED** [review]

**Comment by mchaarawi** [comment]
> LGTM. Though, I'm curious what `dfuse_hl` was used for previously?

it's my original implementation of high level fuse driver that guided at first the DFS implementation / API.
when we started working on dfuse (low level), and that stabilized over time, we stopped testing/maintaining dfuse_hl.

**Pull request merged**